pkg/cli/admin/ocpcertificates/regeneratemco/updatesecrets.go

```diff
@@ -0,0 +1,105 @@
+package regeneratemco
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/vincent-petithory/dataurl"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+func (o *RegenerateMCOOptions) RunUserDataUpdate(ctx context.Context) error {
+	return o.updateSecrets(ctx)
+}
+
+func (o *RegenerateMCOOptions) updateSecrets(ctx context.Context) error {
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	clientset, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	// Get the updated CA cert
+	mcoSecrets := clientset.CoreV1().Secrets(mcoNamespace)
+	mcsSecret, err := mcoSecrets.Get(ctx, newMCSCASecret, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot read MCS secret: %w", err)
+	}
+
+	caCert, exists := mcsSecret.Data[corev1.TLSCertKey]
+	if !exists {
+		return fmt.Errorf("could not find MCS CA cert at %s after rotation", corev1.TLSCertKey)
+	}
+
+	// Get user-data-secret
+	mapiSecrets := clientset.CoreV1().Secrets(mapiNamespace)
+	secretList, err := mapiSecrets.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot list MAO secrets: %w", err)
+	}
+	for _, secret := range secretList.Items {
+		// These two are managed by the MCO but unused. Skip them since the MCO will write them back.
+		if secret.Name == mcoManagedWorkerSecret || secret.Name == mcoManagedMasterSecret {
+			continue
+		}
+
+		// These secrets don't really have a label or not, so the determining factor is if they:
+		// 1. have a userData field
+		// 2. is an ignition config
+		userData, exists := secret.Data[userDataKey]
+		if !exists {
+			continue
+		}
+		// userData is an ignition config. To save the effort of multiple-version parsing, just parse it as a json
+		var userDataIgn interface{}
+		if err = json.Unmarshal(userData, &userDataIgn); err != nil {
+			return fmt.Errorf("failed to unmarshal decoded user-data to json (secret %s): %w", secret.Name, err)
+		}
+
+		_, isIgn, err := unstructured.NestedMap(userDataIgn.(map[string]interface{}), ignFieldIgnition)
+		if !isIgn {
+			// Didn't find ignition in user-data, warn but continue
+			fmt.Fprintf(o.IOStreams.Out, "Unable to find ignition in user-data, skipping secret %s\n", secret.Name)
+			continue
+		}
+
+		caSlice, isSlice, err := unstructured.NestedFieldNoCopy(userDataIgn.(map[string]interface{}), ignFieldIgnition, ignFieldSecurity, ignFieldTLS, ignFieldCA)
```

Comment:
```suggestion
		caSlice, isSlice, err := unstructured.NestedFieldNoCopy(userDataIgn.(map[string]interface{}), ignCAPath...)
```
where
```
ignCAPath = []string{ignFieldIgnition, "security", "tls",  "certificateAuthorities"}
```
defined above instead of the constants